### ISSUE TITLE
Rule proposal: The "model" property should contain a "prop" property and an "event" property validly.

### ISSUE THREAD

**Issue by resessh** [vuejs/eslint-plugin-vue#824]
**Please describe what the rule should do:**

This rule requires `prop` and `event` properties in `model` property.

**What category should the rule belong to?**

- [x] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<script>
export default {
  model: {
    prop: 'someValue',
    // Needs 'event' property.
  },
};
</script>
```

```vue
<script>
export default {
  model: {
    props: 'someValue', // Needs 'prop' property.
    event: 'some-event',
  },
};
</script>
```

**Additional context**

1. If settings is missing, v-model uses `value` as the prop and `input` as the event. But both settings should be required when use `model` property.
2. Miss-typing with property name is easy to occur and founding the mistake is much harder.
